lib/heuristics/concerns/periodic_end_phase.rb

```diff
@@ -423,12 +458,16 @@ def reaffect_prohibiting_partial_assignment
       break if adapted_still_removed.empty? || (adapted_still_removed - banned).empty?
 
       while most_prio_and_frequent.empty?
-        most_prioritary = (adapted_still_removed - banned).group_by{ |removed| @services_data[removed.first][:raw].priority }.min_by{ |priority, _set| priority }[1]
-        most_prio_and_frequent = most_prioritary.group_by{ |removed| removed[1] }.max_by{ |visits_number, _set| visits_number }[1]
+        most_prioritary = (adapted_still_removed - banned).group_by{ |removed|
```

Comment:
If I understand it correctly, it doesn't apply the rule when the block is chained. I decide to leave it to the automation, though since it would be loosing battle to fix them manually :D

Reply:
If I understand it correctly, it doesn't apply the rule when the block is chained. I decide to leave it to the automation, though since it would be loosing battle to fix them manually :D